### ISSUE TITLE
env:import silently failing (leaving user in the dark) any time .env includes reserved environment variables > Output a warning.

### ISSUE THREAD

**Issue by evolve2k** [netlify/cli#5574]
Running env:import silently fails when .env includes reserved environment variables, in my case SITE_NAME.

I made a .env and ran $ netlify env:import .env
Despite multiple attempts I could not get the command to work, leading me down a garden path of attempted other fixes.
An explicit error message would turn a 2 hour random walk into a quick fix 🥺.

The command should return an explicit fail and provide the same warning that is shown in the visual interface.  
Add an error message that is similar to what's in the web validation message, something like:

>  env:import failed. Reason: SITE_NAME is a reserved environment variable. Remove reserved variables from file and retry.

Issue initially reported in the forums here: https://answers.netlify.com/t/netlify-env-import-env-not-uploading-environment-variables-into-the-linked-netlify-project/87695

### Steps to reproduce

How to reproduce:

1. Add a reserved environment variable to .env for example SITE_NAME=Example and VALID_VAR=Something
2. Run $ netlify import:env:import .env
3. Command will silently complete with no indication of success or failure, but it will have failed
4. Check $ netlify env:import and your variables will not have be added
5. Running the command with $ netlify import:env:import .env --debug provides not further insight, but the error should display whether debug is on or not.

### Configuration

My config:

```
### .env
SITE_NAME=Common Guide
REPO_NAME=commonguide
REPO_URL=[GitHub - CommonGuide/commonguide](https://github.com/CommonGuide/commonguide)
EDIT_URI=edit/main/docs/
THEME_NAME=material
```

```
### Log from running ‘netlify env:import .env --debug’:
branch: main
buildDir: /Users/me/projects/commonguide
configPath: /Users/me/projects/commonguide/netlify.toml
context: dev
env:

    DEPLOY_ID
    BUILD_ID
    REPOSITORY_URL
    NETLIFY_LOCAL
    HEAD
    COMMIT_REF
    CACHED_COMMIT_REF
    PULL_REQUEST
    LANG
    LANGUAGE
    LC_ALL
    GATSBY_TELEMETRY_DISABLED
    NEXT_TELEMETRY_DISABLED

❯ Resolved config
build:
command: mkdocs build
commandOrigin: config
publish: /Users/me/projects/commonguide/site
publishOrigin: config
```

### Environment

  System:
    OS: macOS 11.7.4
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 273.87 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.asdf/installs/nodejs/18.15.0/bin/node
    npm: 9.5.0 - ~/.asdf/plugins/nodejs/shims/npm
  npmGlobalPackages:
    netlify-cli: 13.1.3

**Comment by tinfoil-knight** [comment]
@danez **I wanted to know if the Netlify API was changed recently before pushing a fix.**

The code is throwing an `undefined` value which is why things are failing silently. 

The actual error was of this structure:
```
{
  name: "JSONHTTPError",
  status: 422,
  json: {
    code: 422,
    message: "{\"code\":422,\"msg\":\"SITE_NAME is a reserved key name\",\"error_id\":\"{:request_id=\\u003e\\\"cde27457-eb68-4524-9b03-b1dfc7cedda8\\\"}\"}",
  },
}
```

But, in the code we're accessing `.json.msg` instead of `.json.message.msg`.

https://github.com/netlify/cli/blob/be1f66d277ed88133df92dc579fd72d97edd31e3/src/commands/env/env-import.mjs#L108-L112

Note:
- Use of `.json.msg` was added in this PR: https://github.com/netlify/cli/pull/4751
- There are other sub-commands under the `env` command accessing the same value so they might be failing silently too. I'll check this & try to add tests.

**Comment by danez** [comment]
//cc @jasonbarry

**Comment by jasonbarry** [comment]
Hi @evolve2k -- thanks for reporting, and sorry for the inconvenience this has caused you! I can relate to the frustrations of swallowed errors -- there should definitely be some feedback in the output here. 

We recently migrated all sites to use a new environment variable service, and it turns out that errors from that service are getting wrapped in the error message of the API we proxy through. In short, there wasn't a change to our Netlify API as a whole, but rather a change in site data that is causing nested error messages for endpoints proxied to this new service. 

@tinfoil-knight, I've filed an issue for our backend team to take a look at. I think we should handle it there, rather than in the CLI, for uniformity and so that the fix will work with old versions of the CLI too. Let's leave this issue open though until the issue is resolved.
